docs/explanations/rendering.md

```diff
@@ -0,0 +1,344 @@
+custom applications built on top of napari. For example they can create custom
+UI elements which manipulate parameters which generate new images on the fly. In
+these cases when napari UI is blocked it's not just viewing images that's
+blocked, their whole application is frozen and unusable.
+
+ For all of these reasons a primary design goal for rendering is to make sure
+ napari's GUI thread never blocks.
+
+# Framerate
+
+Most screens refresh at 60Hz. To look and feel fully responsive a GUI
+application should strive to draw at 60Hz as well. If 60Hz cannot be acheived
+rendering as fast as possible is desired. The user experience degrades rapidly
+as the refresh rate gets slower:
+
+| Framerate | Milliseconds | User Experience |
+| --------- | ------------ | --------------- |
+| 60Hz      | 16.7         | Great           |
+| 30Hz      | 33.3         | Good            |
+| 20Hz      | 50           | Acceptable      |
+| 10Hz      | 100          | Bad             |
+| 5Hz       | 200          | Unusable        |
+
+Aside from the average framerate if napari renders even a single frame slowly or
+has a pattern of slow and fast frames is can be annoying. People use the term
+[janky](http://jankfree.org/) to describe applications that have framerate
+irregularity. We want napari's average framerate to be high but we also want
+a consistent framerate.
+
+# Array-like Interface
+
+Napari renders data out of an "array-like" interface. This means napari accepts
+any object that can take `numpy`'s slicing syntax. This is a powerful
+abstraction because almost anything could present this type of interface, but
+the flexibility creates a huge challenge for napari since it needs to work with
+basically any source of data.
+
+With **Dask** or custom code it's possible an array access will result in disk
+or network IO. It's even possible the data does not exist at all and it will be
+computed on-the-fly. In this case the user's code is doing the computation and
+napari has no control or visibility into the computation or how long it will
+take.
+
+In #845 the array access leads to loading data from disk or over the network. In
+#1320 the array access leads to a machine learning calculation with Torch. In
+#1300 the problem is different. There the data is already entirely in memory,
+but it's not chunked. Today that means we transfer 100's of MB to the card in
+one shot and this blocks the UI. We can't allow huge monolithic arrays of data
+in napari. Everything must be broken down into reasonably sized chunks.
+
+# Requirements
+
+To met our goal of never blocking the UI we need to satisfy two requirements:
+
+1. Always break data into chunks. The exact maximum chunk size is TBD.
+2. Never call `asarray` on user data from the GUI thread since we don't know
+   what it will do or how long it will take.
+
+# Render Algorithm
+
+The renderer will intersect the current view with the dataset to determine the
```

Comment:
> The renderer will intersect the current view with the dataset to determine the working set.

I'm not sure I completely follow this sentence

Reply:
I was think geometric interaction but really it doesn't have to be spatial so I changed it to "The renderer will compute the **working set** based on the current view. The
working set is the set of chunks that we need to draw to fully render that
specific view."